### PR TITLE
icinga2_host: A lot of fixes after extensive use in production

### PR DESCRIPTION
**Pull request by t794104**
##### SUMMARY

After 1 year in productioin use we have a bunch of fixes

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

icinga2_host.py

##### ADDITIONAL INFORMATION

After extensive use in production we have multiple fixes for this module. This improves stability, error handling and function reliability

### CODE REVIEW

**Comment by Akasurde** [comment]
@t794104 Thanks for the PR. Could you please add test case for this change [here](https://github.com/ansible/ansible/tree/devel/test/units/modules/monitoring) ? Thanks.

needs_info

**Review by Akasurde — CHANGES_REQUESTED** [review]
Could you please write unit test for this change ? Thanks.

**Comment by t794104** [comment]
Schip it

File: .settings/org.eclipse.core.resources.prefs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+eclipse.preferences.version=1
```
Comment (mattclay) [inline .settings/org.eclipse.core.resources.prefs]: This file should not be included in the PR.

I recommend adding `/.settings` to your global git ignore. See https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore for how to do that.

**Comment by Akasurde** [comment]
Hi @t794104, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.general<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**